src/app/Http/Controllers/Admin/LogController.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Controllers\Controller;
+use Symfony\Component\HttpFoundation\Response;
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+
+class LogController extends Controller
+{
+    public function laravel(Request $request): JsonResponse|Response
+    {
+        $user = $request->user();
+        if (!$user->isAdmin()) {
+            return response()->json(['error' => 'Unauthorized.'], 403);
+        }
+
+        $filePath = storage_path('logs/laravel.log');
+
+        if (file_exists($filePath)) {
+            $fileContent = file_get_contents($filePath);
```

Comment:
Rather than reading the whole file into memory, try reading it in chunks. Then you can use streamDownload instead of just download to stream the file back to the client and never have to worry about out of memory errors if the log file is abnormally large. Take a look at the getNawsDump code for an example.

Reply:
done